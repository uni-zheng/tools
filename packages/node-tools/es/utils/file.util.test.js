import { fileUtil } from './file.util';

const fileMock = {
  './a.js': `let a = 1;`,
  './root.js': `
    import('./b.js');
    import c from './c.js';
  `,
  './b.js': `import('./folder1/e.js');`,
  './c.js': `import('./folder1/d.js');`,
  './folder1/d.js': `import('./e.js');`,
  './folder1/e.js': `const e = 1;`,
};

jest.mock('fs', () => {
  return {
    __esModule: true,
    default: {
      readFileSync: jest.fn(filePath => {
        const path = require('path');

        if (path.isAbsolute(filePath)) {
          filePath = `./${path.relative(process.cwd(), filePath)}`;
        }
        
        if (fileMock[filePath]) {
          return fileMock[filePath];
        }
        else {
          throw 'no such file or directory';
        }
      }),
    },
  };
});

describe('fileUtil', () => {
  describe('readFile', () => {
    it('wroks', () => {

      const file = fileUtil.read('./a.js');

      expect(file).toBe('let a = 1;');
    });
  });

  describe('getDependents', () => {
    it('works', () => {
      const file = fileUtil.getDependents('./root.js');

      expect(file).toEqual(
        ['./b.js', './c.js'],
      );
    });
  });

  describe('getDependentGraph', () => {
    it('works', () => {
      const dependentGraph = fileUtil.getDependentGraph('./root.js');

      expect(dependentGraph).toMatchSnapshot();
    });
  });
});
