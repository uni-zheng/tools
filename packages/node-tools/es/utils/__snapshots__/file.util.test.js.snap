// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileUtil getDependentGraph works 1`] = `
Graph {
  "adjList": Map {
    "./root.js" => Array [
      "./b.js",
      "./c.js",
    ],
    "./b.js" => Array [
      "./folder1/e.js",
    ],
    "./folder1/e.js" => Array [],
    "./c.js" => Array [
      "./folder1/d.js",
    ],
    "./folder1/d.js" => Array [
      "./folder1/e.js",
    ],
  },
  "isDirected": true,
  "vertices": Array [
    "./root.js",
    "./b.js",
    "./folder1/e.js",
    "./c.js",
    "./folder1/d.js",
  ],
}
`;
